app/models/user.rb

```diff
@@ -70,6 +70,13 @@ def same_as?(user)
     id == user.id
   end
 
+  # Does the user own a given post?
+  # @param post [Post] post to check
+  # @return [Boolean] check result
+  def owns?(post)
```

Comment:
Not blocking approval, just commenting in case anyone can think of an improvement for this method name.

The meaning of the method is immediately apparent when reading this definition, but if only seeing the method name in use, it could be misinterpreted as meaning author (so a question author might not be expected to be the owner of all of the answers). I don't like any of the names I can think of, but I wonder if anyone can think of something better:
- recursively_owns?(post)
- can_comment_under?(post)
- is_related_to?(post)

Reply:
Yeah, deciding how to name this method was a pain - I opted for `owns?` as we refer to such checks as "post owner", but agreed that the "or the parent, if any" is a good idea to reflect.

Couldn't think of a better name and still can't, though. `recursively_owns?` looks good, but it isn't clear what "recursively" means at a glance. `owner_or_op?` looks good to me too (`is_` is not idiomatic, as I've learned from one of the discussions with @ArtOfCode-) as long as readers can aprse that "op" stands for "original poster" (I can essily, not aure about the others)

Reply:
Thanks for the idiom guidance - I'll bear it in mind.

If the comment rate limit is going to be applied even to a user's own posts and answers to them, does this method become redundant? That would avoid having to name it...

Reply:
What about `owns_post_or_parent?` Bit longer but makes it clear what it does.

Reply:
Yup, looks *perfect* to me - I was looking for something exactly like that, much obliged!